internal/internal_task_pollers_test.go

```diff
@@ -54,3 +59,79 @@ func TestLocalActivityPanic(t *testing.T) {
 	assert.Contains(t, perr.StackTrace(), "panic")
 	assert.Contains(t, perr.StackTrace(), t.Name(), "should mention the source location of the local activity that panicked")
 }
+
+func TestActivityTaskPollerHandlesPanics(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	service := workflowservicetest.NewMockClient(ctrl)
+	service.EXPECT().PollForActivityTask(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(_ context.Context, _ *shared.PollForActivityTaskRequest, opts ...yarpc.CallOption) (*shared.PollForActivityTaskResponse, error) {
+		panic("oh no")
+	})
+	workerStopChannel := make(chan struct{}, 1)
+	activityPoller := newActivityTaskPoller(nil, service, "test", workerExecutionParameters{
+		TaskList: "tasklist",
+
+		WorkerStopChannel: workerStopChannel,
+		WorkerOptions: WorkerOptions{
+			Identity: "identity",
+			Logger:   zaptest.NewLogger(t),
+		},
+	})
+
+	result, err := activityPoller.PollTask()
+
+	assert.Nil(t, result)
+	var panicErr *PanicError
+	assert.ErrorAs(t, err, &panicErr)
+	assert.Equal(t, "oh no", panicErr.value)
+}
+
+func TestActivityTaskPollerHandlesCancel(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	service := workflowservicetest.NewMockClient(ctrl)
+	workerStopChannel := make(chan struct{}, 1)
+	activityPoller := newActivityTaskPoller(nil, service, "test", workerExecutionParameters{
+		TaskList: "tasklist",
+
+		WorkerStopChannel: workerStopChannel,
+		WorkerOptions: WorkerOptions{
+			Identity: "identity",
+			Logger:   zaptest.NewLogger(t),
+		},
+	})
+	service.EXPECT().PollForActivityTask(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(_ context.Context, _ *shared.PollForActivityTaskRequest, opts ...yarpc.CallOption) (*shared.PollForActivityTaskResponse, error) {
+		workerStopChannel <- struct{}{}
+		blockForever := make(chan struct{}, 1)
+		<-blockForever
```

Comment:
should still be something that's cleared out after the test is done, otherwise it leaks test goroutines (we _should_ be using [goleak](https://pkg.go.dev/go.uber.org/goleak), but there are a number of other test leaks at the moment and we haven't spent time on that yet).

if it's only called once, it's very easy to just move this outside the `DoAndReturn` and defer a `close(block)`.